src/tests/java/picard/sam/testers/SamFileTester.java

```diff
@@ -340,4 +342,8 @@ private File createInputFile() {
     public SamReader getInput() {
         return samRecordSetBuilder.getSamReader();
     }
+
+    public void setRecordsNeedSorting(final boolean recordsNeedSorting) {
```

Comment:
So I think the "right" option here is also fairly ugly, so I'll leave it up to you guys whether or not you want to implement it.  It would require declaring `SamFileTester` like this:

```java
class SamFileTester<T extends SamFileTester<T>> {
    public T setRecordsNeedSorting(final boolean recordsNeedSorting) {
        this.recordsNeedSorting = recordsNeedSorting;
        return (T) this;
    }
}
```

And then having subclasses do things like:

```java
class MdSamFileTester extends SamFileTester<MdSamFileTester> { ... }
```

This also means that instantiating the base class becomes ugly: `new SamFileTester<SamFileTester>()`.  You could of course overcome this by providing a static factory method, or having an abstract base class with the generic crap in it and an empty extension class that's instantiated.